Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -237,3 +293,48 @@ pickWaypoints waypoints = do
 safeMod :: Int -> Int -> Int
 _ `safeMod` 0 = 0
 a `safeMod` b = a `mod` b
+
+mkDriverFee ::
+  ( MonadReader r m,
+    MonadGuid m,
+    HasField "driverPaymentCycleBuffer" r NominalDiffTime,
+    HasField "driverPaymentCycleDuration" r NominalDiffTime,
+    HasField "driverPaymentCycleStartTime" r NominalDiffTime
+  ) =>
+  UTCTime ->
+  Id DP.Driver ->
+  Money ->
+  m DF.DriverFee
+mkDriverFee now driverId totalAmount = do
+  id <- generateGUID
+  cycleStartCfg <- asks (.driverPaymentCycleStartTime)
+  cycleDurationCfg <- asks (.driverPaymentCycleDuration)
+  bufferDurationCfg <- asks (.driverPaymentCycleBuffer)
+  let startTime = addUTCTime cycleStartCfg (UTCTime (utctDay now) (secondsToDiffTime 0))
+      endTime = addUTCTime cycleDurationCfg startTime
+  return $
+    DF.DriverFee
+      { payBy = addUTCTime bufferDurationCfg endTime,
+        status = DF.ONGOING,
+        createdAt = now,
+        updatedAt = now,
+        ..
+      }
+
+mkPendingPaymentProcessingKey :: UTCTime -> Text
```

Comment:
shouldn't these keys include driverId?

Reply:
This is to check if the specific job is created in scheduler at given time. The job notifies all drivers which has their payment pending. This isn't driver specific.